extensions/gson/src/main/java/io/jsonwebtoken/io/GsonSerializer.java

```diff
@@ -0,0 +1,43 @@
+package io.jsonwebtoken.io;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.jsonwebtoken.lang.Assert;
+import io.jsonwebtoken.lang.Strings;
+
+public class GsonSerializer<T> implements Serializer<T> {
+
+    static final Gson DEFAULT_GSON = new GsonBuilder().disableHtmlEscaping().create();
+    private Gson gson;
+
+    @SuppressWarnings("unused") //used via reflection by RuntimeClasspathDeserializerLocator
+    public GsonSerializer() {
+        this(DEFAULT_GSON);
+    }
+
+    @SuppressWarnings("WeakerAccess") //intended for end-users to use when providing a custom gson
+    public GsonSerializer(Gson gson) {
+        Assert.notNull(gson, "gson cannot be null.");
+        this.gson = gson;
+    }
+
+    @Override
+    public byte[] serialize(T t) throws SerializationException {
+        Assert.notNull(t, "Object to serialize cannot be null.");
+        //Gson never throws any serialization exception 
+        return writeValueAsBytes(t);
```

Comment:
Per the other implementations (Jackson and org.json), this call is wrapped in a try/catch block, and if any `Exception` is thrown, it is wrapped and re-thrown as a `io.jsonwebtoken.io.SerializationException`.  To ensure runtime consistency, we need to do that here too, i.e.

```java
@Override
public byte[] serialize(T t) throws SerializationException {
    Assert.notNull(t, "Object to serialize cannot be null.");
    try {
        return writeValueAsBytes(t);
    } catch (Exception e) {
        String msg = "Unable to serialize object: " + e.getMessage();
        throw new SerializationException(msg, e);
    }
}
```